### PR TITLE
Refactor(plugins): Move jinja filter code for arista.avd.generate_route_target to PyAVD

### LINKED ISSUE #4100: Refactor(plugins): Move jinja filter code for arista.avd.generate_route_target to PyAVD
### Enhancement summary

Move jinja filter code for arista.avd.generate_route_target to PyAVD.

### Which component of AVD is impacted

plugins

### Use case example

.

### Describe the solution you would like

The code and unit tests for generate_route_target filter should be moved to pyavd/j2filters and imported inside plugins/filter using pyavd wrapper.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Move jinja filter code for arista.avd.generate_route_target to PyAVD

## Related Issue(s)

Fixes #4100 

## Component(s) name

Filter
- `generate_route_target`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/tests/pyavd/j2filters/test_generate_route_target.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import pytest
+from pyavd.j2filters.generate_route_target import generate_route_target
+
+
+class TestEsiManagementFilter:
+    @pytest.mark.parametrize("esi_short, route_target", [("0303:0202:0101", "03:03:02:02:01:01"), (None, None), ("ESI_SHORT", "ES:I_:SH:OR"), ("", "")])
```
Comment (MaheshGSLAB) [inline python-avd/tests/pyavd/j2filters/test_generate_route_target.py]:
```suggestion
    @pytest.mark.parametrize("esi_short, route_target", [("0303:0202:0101", "03:03:02:02:01:01"), (None, None), ("ESI_SHORT", "ES:I_:SH:OR"), ("", None)])
```
Can you please check as convert_esi_short_to_route_target_format was having this condition

**Review by MaheshGSLAB — COMMENTED** [review]
One minor suggestion

File: python-avd/tests/pyavd/j2filters/test_generate_route_target.py (hunk 1 above)
Comment (ClausHolbechArista) [inline python-avd/tests/pyavd/j2filters/test_generate_route_target.py]: Please put the test parameters into a constant to make it easier to read. Thanks.

Comment (ClausHolbechArista) [reply python-avd/tests/pyavd/j2filters/test_generate_route_target.py]: I don't think that would work. an empty string given would return an empty list by the re.find so join will return an empty ~list~ *string.

Comment (Vibhu-gslab) [reply python-avd/tests/pyavd/j2filters/test_generate_route_target.py]: Done!

Comment (Vibhu-gslab) [reply python-avd/tests/pyavd/j2filters/test_generate_route_target.py]: *join will return empty `string`.

File: python-avd/tests/pyavd/j2filters/test_generate_route_target.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ESI_SHORT_NONE = None
+ESI_SHORT_EMPTY_STR = ""
+ESI_SHORT_SINGLE_DIGIT = "3"
+
+
+class TestEsiManagementFilter:
+    @pytest.mark.parametrize(
+        "esi_short, route_target",
+        [(ESI_SHORT, "03:03:02:02:01:01"), (ESI_SHORT_NONE, None), (ESI_SHORT_STR, "ES:I_:SH:OR"), (ESI_SHORT_EMPTY_STR, ""), (ESI_SHORT_SINGLE_DIGIT, "")],
+    )
```
Comment (ClausHolbechArista) [inline python-avd/tests/pyavd/j2filters/test_generate_route_target.py]:
```suggestion
ESI_TO_RT_TEST_CASES = [
    # (<esi_short>, <route_target>)
    ("0303:0202:0101", "03:03:02:02:01:01"),
    (None, None),
    ("ESI_SHORT", "ES:I_:SH:OR"),
    ("", ""),
    ("3", ""),
]


class TestEsiManagementFilter:
    @pytest.mark.parametrize("esi_short, route_target", ESI_TO_RT_TEST_CASES)
```

**Review by ClausHolbechArista — APPROVED** [review]

**Review by ClausHolbechArista — APPROVED** [review]

File: python-avd/pyavd/j2filters/generate_route_target.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+import re
+
+
+def generate_route_target(esi_short: str | None) -> str | None:
+    """
+    generate_route_target Transform 3 octets ESI like 0303:0202:0101 to route-target
```
Comment (Shivani-gslab) [inline python-avd/pyavd/j2filters/generate_route_target.py]:
```suggestion
    generate_route_target transforms 3 octets ESI like 0303:0202:0101 to route-target
```

File: python-avd/pyavd/j2filters/generate_route_target.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    Parameters
+    ----------
+    esi : str
+        Short ESI value as per AVD definition in eos_designs
+
+    Returns
+    -------
+    str
```
Comment (Shivani-gslab) [inline python-avd/pyavd/j2filters/generate_route_target.py]:
```suggestion
    str:
```

**Review by Shivani-gslab — CHANGES_REQUESTED** [review]

**Review by Shivani-gslab — APPROVED** [review]
LGTM

**Review by MaheshGSLAB — APPROVED** [review]

**Pull request merged**